### PR TITLE
Added support for loading CSV with WGS84

### PR DESCRIPTION
**Pull request by michalpelka**
Added way to import WGS84. 
Omitted using o3de asset for handling filenames of CSV.

### CODE REVIEW

**Comment by zakmat** [comment]
Interesting addition. The biggest issue I see is that the change introduces a dependency on a development branch of ROS2 Gem. I won't be able to use this Gem in projects that are based on released ROS2 Gems

**Comment by michalpelka** [comment]
Thanks @zakmat. 
This repo is not canonical to o3de. It serves the community as coding examples. In other words, if someone has a need to use a CSV importer to splines, one can use the last code https://github.com/RobotecAI/robotec-o3de-tools/commit/dc7376de73575583045e1a47262bbfa87629d52a.
We can also change the ROS 2 repo from o3de-extras and split georeference into separate Gem.

File: Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp [hunk 1]
```diff
@@ -189,30 +189,61 @@ namespace SplineTools
             const int index_Y = reader.index_of("y");
             const int index_Z = reader.index_of("z");
 
-            const bool isCoordinateCorrect = !(index_X < 0 || index_Y < 0);
+            const int index_Lat = reader.index_of("lat");
+            const int index_Lon = reader.index_of("lon");
+            const int index_Alt = reader.index_of("alt");
+
+            const bool isCoordinateXY = !(index_X < 0 || index_Y < 0);
+            const bool isCoordinateLatLon = !(index_Lat < 0 || index_Lon < 0 || index_Alt < 0);
```
Comment (pijaro) [inline Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp:197]: Unnecessary new variable? Why don't you use `m_isLatLonAlt`?

File: Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp [hunk 2]
```diff
@@ -189,30 +189,61 @@ namespace SplineTools
...
-            for (csv::CSVRow& row : reader)
+            if (isCoordinateXY)
             {
-                AZ::Vector3 point = AZ::Vector3(row[index_X].get<float>(), row[index_Y].get<float>(), 0);
+                m_isLatLonAlt = false;
+                for (csv::CSVRow& row : reader)
+                {
+                    AZ::Vector3 point = AZ::Vector3(row[index_X].get<float>(), row[index_Y].get<float>(), 0);
+
+                    // handle Z column
```
Comment (pijaro) [inline Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp:215]: Nitpick, unnecessary comment.
```suggestion
```

File: Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp [hunk 3]
```diff
@@ -189,30 +189,61 @@ namespace SplineTools
             const int index_Y = reader.index_of("y");
             const int index_Z = reader.index_of("z");
 
-            const bool isCoordinateCorrect = !(index_X < 0 || index_Y < 0);
+            const int index_Lat = reader.index_of("lat");
+            const int index_Lon = reader.index_of("lon");
+            const int index_Alt = reader.index_of("alt");
```
Comment (pijaro) [inline Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp:194]: Naming convention?

```
index_Lat -> indexLat
index_Lon -> indexLon
index_Alt -> indexAlt
```

**Review by pijaro — CHANGES_REQUESTED** [review]
Can we also update the readme?

**Comment by michalpelka** [comment]
@pijaro  added.

**Review by pijaro — APPROVED** [review]

**Pull request merged**